### ISSUE TITLE
undefined is not a function at new nativeURLSearchParams('s=%2B').get('s')

### ISSUE THREAD

**Issue by mattbarton** [jerrybendy/url-search-params-polyfill#34]
I am getting client-side bug reports of the error above, at line 15. So far I have only seen this on Chrome 69 on Windows 10

**Comment by jerrybendy** [comment]
I just tried on Chrome 69 on Windows 10, and this issue is not repeated. Which version do you use? Can you show me how to repeat this issue?

**Comment by mattbarton** [comment]
I'm not sure how to repeat the issue. Here are the bugsnag reports that i'm seeing:

![image](https://user-images.githubusercontent.com/1208892/47025595-cac04100-d131-11e8-89c6-93fcecd0ca2f.png)

**Comment by jerrybendy** [comment]
Can you try to change `new nativeURLSearchParams('s=%2B').get('s')` to `(new nativeURLSearchParams('s=%2B')).get('s')` ? I don't know if it can work or not 😓

**Comment by apflieger** [comment]
Same here. I got the errors from two user agents :

- Samsung Browser 3 on Android (Marshmallow) `Mozilla/5.0 (Linux; Android 6.0.1; SM-G900F Build/LRX22C) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/3.0 Chrome/38.0.2125.102 Mobile Safari/537.36`
- Chrome 71 on Windows 10 `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36`

I haven't reproduced the problem myself.
I just setup the fix you suggested, let's see how it goes.

**Comment by jerrybendy** [comment]
I have no Samsung device, but I can install a Chrome 71 on Windows 10. Can you make an example code to repeat this bug?

**Comment by apflieger** [comment]
It doesn't fix the error, I just got an other one from a Chrome 72 on Windows 10.
By the way, I am on Chrome 72 on Windows 10 and I don't reproduce it. It is more subtle than that. Apparently `get` is undefined on native URLSearchParams sometimes

**Comment by apflieger** [comment]
I'm trying this change
```
self.URLSearchParams ? self.URLSearchParams : null
``` 
to 
```
(self.URLSearchParams && self.URLSearchParams.get) ? self.URLSearchParams : null
```

**Comment by jerrybendy** [comment]
And what's going on?

`URLSearchParams` has no method of `get`, but `URLSearchParams.prototype.get`

**Comment by apflieger** [comment]
Right, I'm bad a javascript. I just got the error so i'm fixing

**Comment by apflieger** [comment]
Now it breaks further down, here https://github.com/jerrybendy/url-search-params-polyfill/blob/master/index.js#L141 With this error 
```
TypeError: Cannot set property URLSearchParams of [object Object] which has only a getter
```
It's the assignment I think

**Comment by jerrybendy** [comment]
Yes, this is a assignment. It will use a `Proxy` to extend the native `URLSearchParams` if browser supports `Proxy` and has a partial `URLSearchParams` support.

And can you give me a code which can reproduce this issue?

**Comment by apflieger** [comment]
I don't know how to reproduce the problem. I get them by our users because we report uncaught errors with a xhr to the server.

**Comment by apflieger** [comment]
I'm using this change (in https://github.com/jerrybendy/url-search-params-polyfill/pull/37) in production for a month now and got no error

**Comment by jerrybendy** [comment]
Awesome! Although I can't understand why browser supports `URLSearchParams` but no `URLSearchParams.prototype.get`  😂

**Comment by apflieger** [comment]
Yeah, I guess it's something else (a script, plugin or whatever) that puts URLSearchParams that has no get method and blocks further assignation at the same time.

**Comment by jerrybendy** [comment]
It makes sence. I have merged your code (#37), and released version `5.0.1`. I think we can close this issue now. 😅
